### DISCUSSION TITLE
Loop bandwidths (symsync) in what units?

### DISCUSSION THREAD

**Discussion by rognskog** [jgaeddert/liquid-dsp#343] [How do I....?] [answered]
Hello,
How does the units and coefficients of the symsync and iirfilt related to actual units?
For example, if I want to set a loop bandwidth of symsync to 1 Hz (given my sample rate obviously), how can I convert that to the correct normalized units for liquid?

**Comment by jgaeddert** [comment, marked as answer]
Good question! All bandwidths are normalized to the sample rate. So, for example, if you are designing a low-pass filter with a normalized cut-off frequency 0.25, this would result in a filter with a pass-band from -0.25 Fs to +0.25 Fs (where Fs is the sampling frequency).

The symsync object tries to respect this, but its fusing symbol timing and carrier frequency/phase offset recovery loops. Unfortunately, the tracking performance of this is heavily dependent upon the input data (namely the modulation scheme and the excess bandwidth factor of the matched filter). In general, though, if you want to have the signal "settle" after e.g. 1,000 symbols, you would set your loop filter bandwidth to be on the order of 0.001.
